### DISCUSSION TITLE
New Branch creation time

### DISCUSSION THREAD

**Discussion by het20012001** [community/community#63483] [Programming Help]
### Select Topic Area

Question

### Body

Is there any syntax available in pygithub library to check the timing when New Branch is created?

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @het20012001 ! We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the GitHub product Discussions. 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by lab596** [comment]
You can use the GitHub API itself to achieve this functionality. Here's a general outline of how you might approach this:

1. Create a Webhook: Set up a webhook for the repository where you want to track new branch creations. You can configure the webhook to trigger on the "Create" event type.

2. Handle Webhook Payload: When the webhook is triggered, your server will receive a payload with information about the event, including details about the created branch.

3. Extract Timestamp: Extract the timestamp from the payload that corresponds to the time the branch was created. The payload will include a timestamp indicating when the event occurred.

4. Process Timestamp: You can then use the extracted timestamp for your tracking or logging purposes.

Here's a basic example using Flask, a popular web framework for Python, to handle GitHub webhooks:

````python
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/webhook', methods=['POST'])
def webhook():
    payload = request.json
    if 'ref_type' in payload and payload['ref_type'] == 'branch':
        branch_name = payload['ref']
        timestamp = payload['created_at']  # Timestamp when the branch was created

        # Here you can process the branch name and timestamp as needed
        print(f"New branch '{branch_name}' created at {timestamp}")

    return jsonify({'message': 'Webhook received'})

if __name__ == '__main__':
    app.run(host='0.0.0.0', port=5000)
````

**Comment by gamal2001** [comment]
you can use the PyGithub create_git_ref function to create a new branch, However, this function does not return the timing when the new branch is created. To get that information, you may need to use the get_commit function on the new branch and access the commit.committer.date attribute. For example:
`new_branch = repo.get_branch(target_branch)
new_commit = repo.get_commit(new_branch.commit.sha)
new_date = new_commit.commit.committer.date
print(new_date)
`
